subsys/net/l2/openthread/openthread.c

```diff
@@ -96,6 +96,18 @@ static struct net_linkaddr *ll_addr;
 
 static struct net_mgmt_event_callback ip6_addr_cb;
 
+#include <power/reboot.h>
+#include <stdio.h>
+int devicedata_get_pskd(const char **);
+const char *pPSKD = OT_JOINER_PSKD;
+static bool joiner_autostart_started;
+static bool joiner_successful;
+__weak int devicedata_get_pskd(const char **pskd)
```

Comment:
Why this is __weak?

Reply:
so it gives a default implementation, which can be easily overridden by application code

Reply:
Sure, but why application would need or want to override this?